### PR TITLE
Refactoring accountWorkflowService, pulling it up to the service layer to make adding email/phone easier.

### PR DESCRIPTION
**Pull request by alxdarksage**
On an update it will be possible to add an email address (or phone number). When that happens we'll need to send the email verification email for studies that are configured to send it, and that in turn is easier if we refactor to call the AccountWorkflowService from other services, and not from the AccountDao. So this refactor does that in a first check in.

### CODE REVIEW

File: test/org/sagebionetworks/bridge/hibernate/HibernateAccountDaoTest.java [hunk 1]
```diff
@@ -723,17 +690,12 @@ public void createAccountSuccess() {
...
-        verify(mockAccountWorkflowService, never()).sendEmailVerificationToken(any(), any(), any());
     }
 
     @Test
     public void createAccountVerifyEmail() {
-        // Most of this is tested in createAccountSuccess(). Just test email workflow.
-        String accountId = dao.createAccount(STUDY, makeValidGenericAccount(), true);
-        verify(mockAccountWorkflowService).sendEmailVerificationToken(STUDY, accountId, EMAIL);
-
+        dao.createAccount(STUDY, makeValidGenericAccount());
```
Comment (DwayneJengSage) [inline test/org/sagebionetworks/bridge/hibernate/HibernateAccountDaoTest.java]: Now that we don't have verify email in the AccountDao, is this redundant with createAccountSuccess()?

File: app/org/sagebionetworks/bridge/services/AuthenticationService.java [hunk 2]
```diff
@@ -305,7 +305,8 @@ public void verifyEmail(EmailVerification verification) {
         checkNotNull(verification);
 
         Validate.entityThrowingException(EmailVerificationValidator.INSTANCE, verification);
-        accountDao.verifyEmail(verification);
+        Account account = accountWorkflowService.verifyEmail(verification);
+        accountDao.verifyEmail(account);
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/services/AuthenticationService.java:309]: It looks like we don't have unit tests for a lot of the methods in AuthenticationService. Should we add them?

File: app/org/sagebionetworks/bridge/services/ParticipantService.java [hunk 3]
```diff
@@ -260,10 +268,18 @@ public IdentifierHolder createParticipant(Study study, Set<Roles> callerRoles, S
...
+        account.setStatus(sendVerifyEmail ? AccountStatus.UNVERIFIED : AccountStatus.ENABLED);
+        
+        String accountId = accountDao.createAccount(study, account);
+
         externalIdService.assignExternalId(study, participant.getExternalId(), account.getHealthCode());
         optionsService.setAllOptions(study.getStudyIdentifier(), account.getHealthCode(), options);
+        // send verify email
+        if (sendVerifyEmail) {
+            accountWorkflowService.sendEmailVerificationToken(study, accountId, account.getEmail());
+        }
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/services/ParticipantService.java:282]: Is it worth adding unit tests for (1) requestSendVerifyEmail=false (2)requestSendVerifyEmail=true and (3) requestSendVerifyEmail=true but isEmailVerificationEnabled=false?

File: test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java [hunk 4]
```diff
@@ -938,21 +933,6 @@ public void throwLimitExceededException() {
         participantService.createParticipant(STUDY, CALLER_ROLES, PARTICIPANT, false);
     }
     
-    private void verifyStatusCreate(Set<Roles> callerRoles) {
-        mockHealthCodeAndAccountRetrieval();
-        
-        StudyParticipant participant = new StudyParticipant.Builder().copyOf(PARTICIPANT)
-                .withStatus(AccountStatus.ENABLED).build();
```
Comment (DwayneJengSage) [inline test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java:945]: Why were these tests deleted?

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

File: app/org/sagebionetworks/bridge/services/AuthenticationService.java (hunk 2 above)
Comment (alxdarksage) [reply app/org/sagebionetworks/bridge/services/AuthenticationService.java:309]: Between AuthenticationServiceTest and AuthenticationServiceMockTest, all the methods are covered except verifyEmail(...), so I added a couple of tests to the mock test for that.

File: app/org/sagebionetworks/bridge/services/ParticipantService.java (hunk 3 above)
Comment (alxdarksage) [reply app/org/sagebionetworks/bridge/services/ParticipantService.java:282]: Added 4 tests specifically for this. Only the case where email verification is enabled, and the caller wants email verification, calls the workflow service (and account status is tied 1:1 to this, account is only unverified if we're going to send the email).

File: test/org/sagebionetworks/bridge/hibernate/HibernateAccountDaoTest.java (hunk 1 above)
Comment (alxdarksage) [reply test/org/sagebionetworks/bridge/hibernate/HibernateAccountDaoTest.java]: Yes it is, removed it.

File: test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java (hunk 4 above)
Comment (alxdarksage) [reply test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java:945]: These tests were supposed to verify that status was not changed, but it is now changed in the service (and that is now verified in other tests, like those that call verifyStatusUpdate... which looks at impact of roles on whether or not you can change the status, and the tests you just asked me to add to test calls to accountWorkflowService).

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**